jbpm-designer-client/src/main/resources/org/jbpm/designer/public/js/documentation/docstructure.js

```diff
@@ -0,0 +1,659 @@
+                var innerParts = nextPart.split("=");
+                if(innerParts[0].startsWith("[din]")) {
+                    var fromPart = innerParts[0].slice(5, innerParts[0].length);
+                    innerParts.shift(); // removes the first item from the array
+                    var escapedp = innerParts.join('=').replace(/\#\#/g , ",");
+                    escapedp = escapedp.replace(/\|\|/g , "=");
+
+                    retVal += "(Data Input) " + fromPart + " is given value " + escapedp + ", ";
+                } else if(innerParts[0].startsWith("[dout]")) {
+                    var fromPart = innerParts[0].slice(6, innerParts[0].length);
+                    innerParts.shift(); // removes the first item from the array
+                    var escapedp = innerParts.join('=').replace(/\#\#/g , ",");
+                    escapedp = escapedp.replace(/\|\|/g , "=");
+
+                    retVal += "(Data Output) " + fromPart + " is given value " + escapedp + ", ";
+                } else {
+                    // for custom tasks we need to deal with no definition
+                    var fromPart = innerParts[0];
+                    innerParts.shift(); // removes the first item from the array
+                    var escapedp = innerParts.join('=').replace(/\#\#/g , ",");
+                    escapedp = escapedp.replace(/\|\|/g , "=");
+
+                    retVal += "(Data Input) " + fromPart + " is given value " + escapedp + ", ";
+                }
+            } else if(nextPart.indexOf("->") > 0) {
+                var innerParts = nextPart.split("->");
+
+                if(innerParts[0].startsWith("[din]")) {
+                    var fromPart = innerParts[0].slice(5, innerParts[0].length);
+
+                    retVal += "(Data Input) " + fromPart + " is mapped to " + innerParts[1] + ", ";
+                } else if(innerParts[0].startsWith("[dout]")) {
+                    var fromPart = innerParts[0].slice(6, innerParts[0].length);
+
+                    retVal += "(Data Output) " + fromPart + " is mapped to " + innerParts[1] + ", ";
+                }
+            } else {
+                // default to equality
+                if(innerParts[0].startsWith("[din]")) {
+                    var fromPart = innerParts[0].slice(5, innerParts[0].length);
+
+                    retVal += "(Data Input) " + fromPart + " is given value  , ";
+                } else if(innerParts[0].startsWith("[dout]")) {
+                    var fromPart = innerParts[0].slice(5, innerParts[0].length);
+
+                    retVal += "(Data Output) " + fromPart + " is given value  , ";
+                }
+            }
+        }
+
+        if(retVal.length > 0) {
+            return retVal.substring(0, retVal.length - 2);
+        } else {
+            return propvalue;
+        }
+    } else {
+        return propvalue;
+    }
+}
+
+function presentPropertyName(rawprop) {
+    if(rawprop == "assignmentsview") {
```

Comment:
These are all static values so why not simply replace the giant if/else if with a associative array/map lookup i.e. define somehwere ```var propertyNames = {"assignmentsview": "Assignments Count", "script_language": "Script Language", ...}``` and then this method can simply look up using return ```propertyNames[rawprop]```.